File: Hooligans-bon_plan_web/src/Entity/Bonplan.php
<?php

namespace App\Entity;

use App\Repository\BonplanRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: BonplanRepository::class)]
class Bonplan
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name:"id_bonplan")]
    private ?int $id = null;

    #[ORM\Column(length: 50)]
    #[Assert\NotBlank(message:"Le nom du bon plan est obligatoire !")]
    private ?string $nom_bonplan = null;

    #[ORM\Column(length: 50)]
    private ?string $gouvernorat = null;

    #[ORM\Column(length: 50)]
    private ?string $ville = null;

    #[ORM\Column(length: 255)]
    private ?string $rue = null;

    #[ORM\Column(length: 255)]
    private ?string $description = null;

    #[ORM\Column(length: 50)]
    private ?string $type = null;

    #[ORM\Column(length: 20)]
    private ?string $etat = null;

    #[ORM\Column(length: 255)]
    private ?string $image = null;

    #[ORM\ManyToOne]
    #[ORM\JoinColumn(nullable: false,name:"id_user",referencedColumnName:"id_user")]
    private ?User $user = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomBonplan(): ?string
    {
        return $this->nom_bonplan;
    }

    public function setNomBonplan(string $nom_bonplan): self
    {
        $this->nom_bonplan = $nom_bonplan;

        return $this;
    }

    public function getGouvernorat(): ?string
    {
        return $this->gouvernorat;
    }

    public function setGouvernorat(string $gouvernorat): self
    {
        $this->gouvernorat = $gouvernorat;

        return $this;
    }

    public function getVille(): ?string
    {
        return $this->ville;
    }

    public function setVille(string $ville): self
    {
        $this->ville = $ville;

        return $this;
    }

    public function getRue(): ?string
    {
        return $this->rue;
    }

    public function setRue(string $rue): self
    {
        $this->rue = $rue;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getUser(): ?User
    {
        return $this->user;
    }

    public function setUser(?User $user): self
    {
        $this->user = $user;

        return $this;
    }
}

File: src/models/BonPlan.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package models;

import models.User;

/**
 *
 * @author azizh
 */
public class BonPlan {
   private int id_bonplan;
   private String nom_bonplan;
   private String gouvernorat;
    private String ville;
   private String rue;
   private String type;
   private String etat="en attente";
   private String image;
     private String description;
   private User user;
   
   

   
  
    // constructeur qui contient tous les params avec etat

    public BonPlan(int id_bonplan, String nom_bonplan, String gouvernorat, String ville, String rue, String type, String image, String description, User user) {
        this.id_bonplan = id_bonplan;
        this.nom_bonplan = nom_bonplan;
        this.gouvernorat = gouvernorat;
        this.ville = ville;
        this.rue = rue;
        this.type = type;
        this.image = image;
        this.description = description;
        this.user = user;
    }

    // constructeur qui contient tous les params sauf l id_bonplan

    public BonPlan(String nom_bonplan, String gouvernorat, String ville, String rue, String type, String image, String description, User user) {
        this.nom_bonplan = nom_bonplan;
        this.gouvernorat = gouvernorat;
        this.ville = ville;
        this.rue = rue;
        this.type = type;
        this.image = image;
        this.description = description;
        this.user = user;
    }
  
    
    // constructeur qui contient tous les params sauf l id_bonplan et l'etat
    public BonPlan(String nom_bonplan, String gouvernorat, String ville, String rue, String image, String description, User user) { 
        this.nom_bonplan = nom_bonplan;
        this.gouvernorat = gouvernorat;
        this.ville = ville;
        this.rue = rue;
        this.image = image;
        this.description = description;
        this.user = user;
    }

    // constructeur non parameteré
    public BonPlan() {
    }
    
    // getters and setters 

    public int getId_bonplan() {
        return id_bonplan;
    }

    public void setId_bonplan(int id_bonplan) {
        this.id_bonplan = id_bonplan;
    }

    public String getNom_bonplan() {
        return nom_bonplan;
    }

    public void setNom_bonplan(String nom_bonplan) {
        this.nom_bonplan = nom_bonplan;
    }

    public String getGouvernorat() {
        return gouvernorat;
    }

    public void setGouvernorat(String gouvernorat) {
        this.gouvernorat = gouvernorat;
    }

    public String getVille() {
        return ville;
    }

    public void setVille(String ville) {
        this.ville = ville;
    }

    public String getRue() {
        return rue;
    }

    public void setRue(String rue) {
        this.rue = rue;
    }



    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }
    

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }


    
    // Affichage 

    @Override
    public String toString() {
        return "BonPlan{" + "id_bonplan=" + id_bonplan + ", nom_bonplan=" + nom_bonplan + ", gouvernorat=" + gouvernorat + ", ville=" + ville + ", rue=" + rue + ", type=" + type + ", etat=" + etat + ", image=" + image + ", description=" + description + ", user=" + user + '}';
    }

 

    

    
    
    
}